### ISSUE TITLE
Cilium nodes failing health check with endpoints

### ISSUE THREAD

**Issue by karanrn** [cilium/cilium#18294]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

We have deployed EKS cluster in us-west-2 region and installed Cilium (v1.11) as CNI via helm to not use ENI mode (instead use overlay network)
```
helm install cilium cilium/cilium --version 1.11.0 \
  --namespace kube-system \
  --set egressMasqueradeInterfaces=eth0 \
  --set nodeinit.enabled=true
```

And when we check cilium status or cilium health status, it shows nodes are not reachable via endpoints.
We followed/observed below steps - 
1. Connectivity test passed
2. Checked IPTables and it looks good with no DROP or REJECT rules for Cilium packets and also compared IPTables of nodes, we see no difference between them
3. Checked datapath state of packet from a node using `cilium monitor --type drop` and we see no packet being dropped
4. Monitored for packets if they are dropped elsewhere in linux stack using `cilium monitor -v | grep drop`
5. Monitored for packets for endpoint on another node using `cilium monitor --to 258` but no packets are being sent out

Strange thing, we see only one node can reach all the nodes via endpoint, we checked IPTables of this perfect node with problematic nodes and we see no difference. 

We did reinstallation too but no luck.
There is similar issue but with ENI enabled so had to open an new issue #17280

Are we doing something wrong here or missing something?

### Cilium Version

1.11.0

### Kernel Version

Linux ip-10-XX-XX-XX.us-west-2.compute.internal 5.4.149-73.259.amzn2.x86_64 #1 SMP Mon Sep 27 12:48:12 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

{
  "clientVersion": {
    "major": "1",
    "minor": "22",
    "gitVersion": "v1.22.3",
    "gitCommit": "c92036820499fedefec0f847e2054d824aea6cd1",
    "gitTreeState": "clean",
    "buildDate": "2021-10-27T18:41:28Z",
    "goVersion": "go1.16.9",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "serverVersion": {
    "major": "1",
    "minor": "21+",
    "gitVersion": "v1.21.2-eks-06eac09",
    "gitCommit": "5f6d83fe4cb7febb5f4f4e39b3b2b64ebbbe3e97",
    "gitTreeState": "clean",
    "buildDate": "2021-09-13T14:20:15Z",
    "goVersion": "go1.16.5",
    "compiler": "gc",
    "platform": "linux/amd64"
  }
}

### Sysdump

[cilium-sysdump-20211217-044935.zip](https://github.com/cilium/cilium/files/7732157/cilium-sysdump-20211217-044935.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by clementnuss** [comment]
Hi @karanrn 

we've observed an issue (which might be linked to yours) on our cluster where with Cilium v1.11.0 the agent doesn't update the health check IP when a node/cilium agent changes.
as a consequence, cilium-health tries to reach the old pod/node/agent ip address, which it cannot, so it thinks the node isn't reachable via endpoint.

can you try to run `cilium-health status` on one of the nodes, then check and see whether the IP addresses it cannot reach are up-to-date with their current definition in the `CiliumNode` resource ?
i.e. on node A, if `cilium-health status` cannot reach IP xxx for node B, go check the CiliumNode resource of node B and check the `.spec.health` to see if the two IP match

curious to see if this is the same issue as we are encountering !

**Comment by karanrn** [comment]
Hi @clementnuss

Nodes in our cluster are reachable via IP but are not reachable via endpoints.
And I checked the IP of the health endpoint of Node B, it is the same as the not reachable endpoint IP from Node A (it is updated).

**Comment by brb** [comment]
@clementnuss this seems to be yet another instance of your issue - https://github.com/cilium/cilium/issues/18302

**Comment by aanm** [comment]
cc @christarazi it seems it is the same as https://github.com/cilium/cilium/issues/18302

**Comment by christarazi** [comment]
Closed as it's the same as #18302
